lightning_toolbox/training/module.py

```diff
@@ -140,10 +144,23 @@ def is_optimizer_active(self, optimizer_idx: int, batch_idx: int, epoch: int) ->
             result = is_active(training_module=self, optimizer_idx=optimizer_idx, batch_idx=batch_idx, epoch=epoch)
         return result
 
+    def remember(self, **kwargs: th.Dict[str, th.Any]) -> None:
+        """
+        You can use this function to remember any particular object that you want to
+        use later on.
+
+        **Note**: The values that are remembered will be forgotten soon after every batch
+        so make sure to save them in another logging mechanism if you want to keep them.
+        Typically, this is implemented using a LoggingCallback.
+        """
+        self.objective.remember(**kwargs)
+
     def forward(self, inputs):
         "Placeholder forward pass for the model"
         if hasattr(self, "model") and self.model is not None:
-            return self.model(inputs)
+            # Also pass the training module to the model so that it can access the objective
```

Comment:
This is an excellent idea. Still, the problem could be that the model you are using does not support a training_module kw_argument, and this would result in an error. To overcome such a problem, you have to wrap the forward method of the model in dypy.dynamic_args_wrapper, so that providing any potentially additional arguments (e.g. training_module here) does not raise an exception, and the model forward can decide to use the information present in training_module, or not by providing a forward function that has training_module in its signature.

Reply:
And now that you are on it, let's change the default forward implementation here so that it takes on `*args, **kwargs` and passes it on to the model's forward function directly.